torch/sparse/__init__.py

```diff
@@ -150,6 +150,46 @@
 """)
 
 
+spmm_reduce = _add_docstr(_sparse.spmm_reduce, r"""
```

Comment:
About the naming of this API, it is sparse-dense matmul by definition, would it make more sense to be named as `spdmm_reduce`?

Reply:
spmm is the right name, mkl's spmm just means that A is CSR and B is dense.

Reply:
Hmm, mkl spmm is actually on two sparse matrices: https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-c/top/blas-and-sparse-blas-routines/inspector-executor-sparse-blas-routines/inspector-executor-sparse-blas-execution-routines/mkl-sparse-spmm.html

On the other hand, the sparse/dense matmul is named something like mkl_sparse_s_mm:
https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-c/top/blas-and-sparse-blas-routines/inspector-executor-sparse-blas-routines/inspector-executor-sparse-blas-execution-routines/mkl-sparse-mm.html